test-data/unit/check-functions.test

```diff
@@ -2125,3 +2125,82 @@ def i() -> List[Union[str, int]]:
     return x
 
 [builtins fixtures/dict.pyi]
+
+[case testAnyStrIncompatibleArguments]
+from typing import TypeVar
+AnyStr = TypeVar('AnyStr', str, bytes)
+def f(x: AnyStr, y: AnyStr) -> None: pass
+def g(x: AnyStr, y: AnyStr, z: int) -> AnyStr: pass
+f('a', 'b')
+f(b'a', b'b')
+f('a', b'b') # E: Type argument 1 of "f" has incompatible value \
+# N: "AnyStr" must be all one type: str or bytes
+g('a', 'b', 1)
+g(b'a', b'b', 1)
+g('a', b'b', 1) # E: Type argument 1 of "g" has incompatible value \
+# N: "AnyStr" must be all one type: str or bytes
+g('a', b'b', 'c') # E: Type argument 1 of "g" has incompatible value \
+# N: "AnyStr" must be all one type: str or bytes \
+# E: Argument 3 to "g" has incompatible type "str"; expected "int"
+
+[case testUnionAnyStrIncompatibleArguments]
+from typing import TypeVar, Union
+AnyStr = TypeVar('AnyStr', str, bytes)
+def f(x: Union[AnyStr, int], y: AnyStr) -> None: pass
+f('a', 'b')
+f(1, 'b')
+f('a', b'b') # E: Type argument 1 of "f" has incompatible value \
+# N: "AnyStr" must be all one type: str or bytes
+
+[case testStarAnyStrIncompatibleArguments]
+from typing import TypeVar, Union
+AnyStr = TypeVar('AnyStr', str, bytes)
+def f(*x: AnyStr) -> None: pass
+def g(x: int, *y: AnyStr) -> None: pass
+def h(*x: AnyStr, y: int) -> None: pass
+f('a')
+f('a', 'b')
+f('a', b'b') # E: Type argument 1 of "f" has incompatible value \
+# N: "AnyStr" must be all one type: str or bytes
+g(1, 'a')
+g(1, 'a', b'b') # E: Type argument 1 of "g" has incompatible value \
+# N: "AnyStr" must be all one type: str or bytes
+h('a', y=1)
+h('a', 'b', y=1)
+h('a', b'b', y=1) # E: Type argument 1 of "h" has incompatible value "object"
```

Comment:
Can you point me in the right direction on how to decode `Callable.arg_kinds`? For this case it is just `[2,3]` which doesn't mean much to me. Can I compare this to `Context.arg_kinds`? In this case it is `[0,0,3]`.

The only other time this error occurs is on line 961 of `check-overloading.test` which I don't think is in scope for this pull request and line 233 of `check-typevar-values.text` which is actually passing an argument with type `object`.